var/spack/repos/builtin/packages/octopus/package.py

```diff
@@ -134,12 +228,113 @@ def install(self, spec, prefix):
         if (spec.satisfies('%apple-clang') or
                 spec.satisfies('%clang') or
                 spec.satisfies('%gcc')):
-            args.extend([
-                'FCFLAGS=-O2 -ffree-line-length-none'
-            ])
+            # In case of GCC version 10, we will have errors because of
+            # argument mismatching. Need to provide a flag to turn this into a
+            # warning and build sucessfully
+            if (spec.satisfies('%gcc@10:')):
+                args.extend([
```

Comment:
I think you could avoid duplicating the flags that are common to all versions.
```
args.extends(['common_flags'])
if ...:
  args.extends(['gcc10_specific_flags'])
```

Reply:
I have addressed that in https://github.com/spack/spack/pull/30321/files/a4dfd5f4f42c0217df7669cda4ea7a4f69faaef1..255d665fae6a686e4ea51f3d2f9c54f55638ef09 .